### ISSUE TITLE
Test guest VM fails to mount host filesystem to /var/dbfit

### ISSUE THREAD

**Issue by MMatten** [dbfit/dbfit#510]
The Vagrant VM build fails as the `chef/centos-7.1` image no longer seems available and Vagrant tells me that the box name is now `bento/centos-7.1`.

As per https://atlas.hashicorp.com/chef/ the boxes seem to have moved to https://atlas.hashicorp.com/bento/.

Updating the `Vagrantfile` to specify `config.vm.box = "bento/centos-7.1"` gets the basic VM up and running but the NFS mounting fails.

The command that tries to mount the host FS on the guest at `/var/dbfit` fails because of the failing command `mount -o vers=4,udp,noatime,nodiratime 192.168.33.1:/home/mmatten/dbfit /var/dbfit` on my host Ubuntu 16.04 LTS. 

The error message being `mount.nfs: an incorrect mount option was specified`.

It seems that the issue is the `udp` option. I _can't_ see this listed on man pages for `mount`. I can't even find a specific Centos 7 man page for `mount`.

Any ideas what's causing Vagrant to construct this mount command with this option?

**Comment by MMatten** [comment]
Ok. UDP is the default (https://www.vagrantup.com/docs/synced-folders/nfs.html#nfs_udp).

I'm not sure why it's failing in my environment but I can override the default in `vagrant_config_defaults.rb` with `nfs_udp: false`.

**Comment by javornikolov** [comment]
Yes, seems Vagrant is not so smart to automatically remove `udp` option in case of nfs 4. Looking at the vagrant explanation on the given link - that would be a general issue I guess (not specific to your environment only) - we may add this `nfs_udp: false` to `vagrant_config_defaults.rb`.

**Comment by MMatten** [comment]
This `bento/centos-7.1` box has issues too. It seem that the kernel headers are missing and so the guest additions won't build and that affects the NFS stuff.

``/tmp/vbox.0/Makefile.include.header:115: *** Error: unable to find the include directory for your current Linux kernel. Specify KERN_INCL=<dire
ctory> and run Make again.  Stop.
Creating user for the Guest Additions.
Creating udev rule for the Guest Additions kernel module.
`

**Comment by javornikolov** [comment]
Do you have the latest version of `vagrant`, `vagrant-vbguest`, `VirtaulBox`? I think the `vbguest` plugin is supposed to download kernel headers if missing. There used to be some issues in past and caveats to workaround them...

In general we should try with more recent CentOS version (`bento/centos-7.2` or maybe `centos/7`).

**Comment by MMatten** [comment]
I've got the very latest Virtualbox. I've got the very latest vagrant. I've only just installed it all on a new Ubuntu 16.04 host. 

I've tried the `bento/centos-7.2` box too. It doesn't seem to boot and Vagrant times out. I can try upping the timeout and seeing if it helps.

I've try the `centos/7` box too.

**Comment by MMatten** [comment]
The `bento/centos-7.2` seems very odd. It's not really a timeout. Vagrant doesn't even appear to be retrying.

```
==> default: Waiting for machine to boot. This may take a few minutes...
    default: SSH address: 127.0.0.1:2222
    default: SSH username: vagrant
    default: SSH auth method: private key
Timed out while waiting for the machine to boot. This means that
Vagrant was unable to communicate with the guest machine within
the configured ("config.vm.boot_timeout" value) time period.

If you look above, you should be able to see the error(s) that
Vagrant had when attempting to connect to the machine. These errors
are usually good hints as to what may be wrong.

If you're using a custom box, make sure that networking is properly
working and you're able to connect to the machine. It is a common
problem that networking isn't setup properly in these boxes.
Verify that authentication configurations are also setup properly,
as well.

If the box appears to be booting properly, you may want
```

**Comment by MMatten** [comment]
The `centos/7` base box is missing the `SATA Controller` disk controller the pre-boot customisations fail. I've added it manually and am continuing with the provisioning.

Perhaps we could attach our extra disk to the IDE Controller that the box does have. Do you know what we'd need to change in the `Vagrantfile`? Just the named controller?

**Comment by MMatten** [comment]
Relating to this - does this project have any S3 space where we could put a stable base box image (at least as a fall back position)?

**Comment by javornikolov** [comment]
Weird... If SATA controller is missing - isn't there SCSI controller at least? Some hints about adding one: 
https://gist.github.com/leifg/4713995

The `centos/7` image seems to be provided by the CentOS maintainers (https://seven.centos.org/2016/10/updated-centos-vagrant-images-available-v1609-01/). It's supposed to work OK, if not - we may report the issues upstream.

**Comment by javornikolov** [comment]
Images for previous CentOS builds are available also here: http://cloud.centos.org/centos/7/vagrant/x86_64/images/

BTW, is your [`vagrant-vbguest`](https://github.com/dotless-de/vagrant-vbguest) plugin version 0.13.0?

**Comment by MMatten** [comment]
In fact I think this base box has a 40gb IDE disk so I'm going to try removing the addition of the extra disk we're trying to adding.

**Comment by MMatten** [comment]
> BTW, is your vagrant-vbguest plugin version 0.13.0?

Yes.

**Comment by MMatten** [comment]
Now issues with MySQL starting. I'll see what messages I can find.

```
==> default:       ================================================================================
==> default:       Error executing action `start` on resource 'service[default :start mysql-default]'
==> default:       ================================================================================
==> default:       
==> default:       Mixlib::ShellOut::ShellCommandFailed
==> default:       ------------------------------------
==> default:       Expected process to exit with [0], but received '1'
==> default:       ---- Begin output of /bin/systemctl --system start mysql-default ----
==> default:       STDOUT: 
==> default:       STDERR: Job for mysql-default.service failed because the control process exited with error code. See "systemctl status mysql-default.service" and "journalctl -xe" for details.
==> default:       ---- End output of /bin/systemctl --system start mysql-default ----
==> default:       Ran /bin/systemctl --system start mysql-default returned 1

```

**Comment by MMatten** [comment]
As mentioned in the Vagrant error messages, running `journalctl -xe` shows: -

```
-- Unit mysql-default.service has begun starting up.
Oct 16 20:28:02 dbfitvm mysqld[9888]: 161016 20:28:02 [Note] /usr/sbin/mysqld (mysqld 5.5.53) starting as process 9888 ...
Oct 16 20:28:02 dbfitvm mysqld[9888]: 161016 20:28:02 [Note] Plugin 'FEDERATED' is disabled.
Oct 16 20:28:02 dbfitvm mysqld[9888]: [68B blob data]
Oct 16 20:28:02 dbfitvm mysqld[9888]: 161016 20:28:02 [ERROR] Can't open the mysql.plugin table. Please run mysql_upgrade to create it.
Oct 16 20:28:02 dbfitvm mysqld[9888]: 161016 20:28:02 InnoDB: The InnoDB memory heap is disabled
Oct 16 20:28:02 dbfitvm mysqld[9888]: 161016 20:28:02 InnoDB: Mutexes and rw_locks use GCC atomic builtins
Oct 16 20:28:02 dbfitvm mysqld[9888]: 161016 20:28:02 InnoDB: Compressed tables use zlib 1.2.3
Oct 16 20:28:02 dbfitvm mysqld[9888]: 161016 20:28:02 InnoDB: Using Linux native AIO
Oct 16 20:28:02 dbfitvm mysqld[9888]: 161016 20:28:02 InnoDB: Initializing buffer pool, size = 128.0M
Oct 16 20:28:02 dbfitvm mysqld[9888]: 161016 20:28:02 InnoDB: Completed initialization of buffer pool
Oct 16 20:28:02 dbfitvm mysqld[9888]: 161016 20:28:02  InnoDB: Operating system error number 13 in a file operation.
Oct 16 20:28:02 dbfitvm mysqld[9888]: InnoDB: The error means mysqld does not have the access rights to
Oct 16 20:28:02 dbfitvm mysqld[9888]: InnoDB: the directory.
Oct 16 20:28:02 dbfitvm mysqld[9888]: InnoDB: File name ./ibdata1
Oct 16 20:28:02 dbfitvm mysqld[9888]: InnoDB: File operation call: 'open'.
Oct 16 20:28:02 dbfitvm mysqld[9888]: InnoDB: Cannot continue operation.
Oct 16 20:28:02 dbfitvm systemd[1]: mysql-default.service: main process exited, code=exited, status=1/FAILURE
Oct 16 20:28:03 dbfitvm systemd[1]: mysql-default.service: control process exited, code=exited status=1
Oct 16 20:28:03 dbfitvm systemd[1]: Failed to start mysql_service[mysql-default].
-- Subject: Unit mysql-default.service has failed
-- Defined-By: systemd
-- Support: http://lists.freedesktop.org/mailman/listinfo/systemd-devel
-- 
-- Unit mysql-default.service has failed.
-- 
-- The result is failed.
```

Some file system permission error!?

**Comment by javornikolov** [comment]
You may try to refresh your cookbooks (`rm -rf Cheffile.lock cookbooks tmp`).

**Comment by javornikolov** [comment]
Have you tried the suggested `mysql_upgrade`?

**Comment by MMatten** [comment]
> Have you tried the suggested mysql_upgrade?

There are many options for the command. I'm not sure where to start. But we need this running from Chef anyway.

**Comment by MMatten** [comment]
> You may try to refresh your cookbooks (rm -rf Cheffile.lock cookbooks tmp).

I'll try using latest versions of other cookbooks.

**Comment by MMatten** [comment]
Ok. Now and error about MySQL client: -

```
==> default: Chef Client failed. 25 resources updated in 05 minutes 48 seconds
==> default: [2016-10-16T20:56:01+00:00] FATAL: Stacktrace dumped to /var/chef/cache/chef-stacktrace.out
==> default: [2016-10-16T20:56:01+00:00] FATAL: Please provide the contents of the stacktrace.out file if you file a bug report
==> default: [2016-10-16T20:56:01+00:00] ERROR: mysql2_chef_gem[default] (dbfit_test::mysql line 1) had an error: Chef::Exceptions::Package: mysql_client_installation_package[default] (/var/chef/cache/cookbooks/mysql2_chef_gem/libraries/provider_mysql2_chef_gem_mysql.rb line 16) had an error: Chef::Exceptions::Package: yum_package[mysql-community-client] (/var/chef/cache/cookbooks/mysql/libraries/mysql_client_installation_package.rb line 18) had an error: Chef::Exceptions::Package: No candidate version available for mysql-community-client

```

**Comment by javornikolov** [comment]
Which cookbook version is that (should be visible in Cheffile.lock). Possibly related issue: https://github.com/chef-cookbooks/mysql/issues/446

About the server: some leads about issues if `SELinux` is enabled, also suggestions to check the log file `/var/log/mysqld.log`.

**Comment by MMatten** [comment]
I can't see anything useful in `/var/chef/cache/chef-stacktrace.out` either.

**Comment by MMatten** [comment]
In Cheffile.lock I can see:

```
mysql (8.0.4)
    mysql2_chef_gem (1.1.0)
      build-essential (>= 0.0.0)
      mariadb (>= 0.0.0)
      mysql (>= 6.0)
```

**Comment by MMatten** [comment]
> About the server: some leads about issues if SELinux is enabled, also suggestions to check the log file /var/log/mysqld.log.

I can't find a `mysqld.log` file anywhere on the guest.

**Comment by javornikolov** [comment]
https://github.com/sinfomicien/mysql2_chef_gem/issues/22

**Comment by MMatten** [comment]
Thanks.

I'll follow that up tomorrow (I have to stop for today now).

I'll try turning down SELinux enforcement too (https://supermarket.chef.io/cookbooks/selinux/versions/0.5.0) tomorrow.

Thanks for the help so far!

**Comment by javornikolov** [comment]
You're welcome :-) Thanks for your efforts to resurrect the test VM setup!

**Comment by MMatten** [comment]
> sinfomicien/mysql2_chef_gem#22

I've not really got anywhere following this thread.

I've tried installing `libmysqlclient-dev` and `mysql-devel` just before `mysql2_chef_gem 'default' do` in `mysql.rb`. I'm not sure why, but using either of `package` or `yum_package` both fail to install the libraries (no candidate versions).

I installed in manually (`sudo yum install `mysql-devel`) but them I'm back to the original problem (`No candidate version available for mysql-community-client`).

Is there some general issue with MySQL in public (Oracle's?) package repos?

**Comment by javornikolov** [comment]
What happens if you do: `sudo yum install mysql-community-client`? Maybe the issue is that it ends up searching for 

Which version of mysql server/client is getting installed on the box? You may try to set specific version in Vagrantfile under mysql, e.g. `version => 5.7`.

I guess the latest versions of mysql - Oracle's packages, are not available in the standard packages for CentOS/RHEL - i.e. it needs extra steps to setup the Oracle yum repo access (https://dev.mysql.com/doc/mysql-yum-repo-quick-guide/en/). I suppose the chef cookbook should be doing that setup behind the scenes.

**Comment by javornikolov** [comment]
BTW, I can see `recipe[selinux::disabled]` mentioned as requirement here: https://github.com/chef-cookbooks/mysql

**Comment by javornikolov** [comment]
Also in the same README: `As of the 8.0 version of this cookbook, configuration of the package repos is now the responsibility of the user`. We used to be on version `6.1.0` as far as I see from Cheffile.lock - so maybe some changes are needed when migrating to v8+.

**Comment by MMatten** [comment]
> What happens if you do: sudo yum install mysql-community-client? Maybe the issue is that it ends up searching for non-existing package.

I'd added the MySQL yum repo on the guest as mentioned at http://dev.mysql.com/doc/mysql-yum-repo-quick-guide/en/.

So when I run `sudo yum install mysql-community-client` I'm offered the 5.7 packages.

But when running via the Chef provisioning it seems to be looking for 5.6.29-2.el7 of mysql-community-client (from the messages following). Is 5.6.x set somewhere in the 



==> default:         ================================================================================
==> default:         Error executing action `install` on resource 'yum_package[mysql-community-client]'
==> default:         ================================================================================
==> default:         
==> default:         Chef::Exceptions::Package
==> default:         -------------------------
==> default:         Version 5.6.29-2.el7 of mysql-community-client not found. Did you specify both version and release? (version-release, e.g. 1.84-10.fc6)
==> default:         
==> default:         Resource Declaration:
==> default:         ---------------------
==> default:         # In /var/chef/cache/cookbooks/mysql/libraries/mysql_client_installation_package.rb
==> default:         
==> default:          18:       package package_name do
==> default:          19:         version package_version if package_version
==> default:          20:         options package_options if package_options
==> default:          21:         action :install
==> default:          22:       end
==> default:          23:     end
==> default:         
==> default:         Compiled Resource:
==> default:         ------------------
==> default:         # Declared in /var/chef/cache/cookbooks/mysql/libraries/mysql_client_installation_package.rb:18:in `block in <class:MysqlClientInstallationPackage>'
==> default:         
==> default:         yum_package("mysql-community-client") do
==> default:           package_name "mysql-community-client"
==> default:           action [:install]
==> default:           retries 0
==> default:           retry_delay 2
==> default:           default_guard_interpreter :default
==> default:           declared_type :package
==> default:           cookbook_name "dbfit_test"
==> default:           flush_cache {:before=>false, :after=>false}
==> default:           version "5.6.29-2.el7"
==> default:         end
==> default:         
==> default:         Platform:
==> default:         ---------
==> default:         x86_64-linux
==> default:         
==> default: [2016-10-17T21:59:48+00:00] INFO: Running queued delayed notifications before re-raising exception
==> default:       
==> default:       ================================================================================
==> default:       Error executing action `create` on resource 'mysql_client_installation_package[default]'
==> default:       ================================================================================
==> default:       
==> default:       Chef::Exceptions::Package
==> default:       -------------------------
==> default:       yum_package[mysql-community-client] (/var/chef/cache/cookbooks/mysql/libraries/mysql_client_installation_package.rb line 18) had an error: Chef::Exceptions::Package: Version 5.6.29-2.el7 of mysql-community-client not found. Did you specify both version and release? (version-release, e.g. 1.84-10.fc6)
==> default:       
==> default:       Resource Declaration:
==> default:       ---------------------
==> default:       # In /var/chef/cache/cookbooks/mysql2_chef_gem/libraries/provider_mysql2_chef_gem_mysql.rb
==> default:       
==> default:        16:           mysql_client 'default' do
==> default:        17:             version new_resource.client_version
==> default:        18:             action :create
==> default:        19:           end
==> default:        20: 
==> default:       
==> default:       Compiled Resource:
==> default:       ------------------
==> default:       # Declared in /var/chef/cache/cookbooks/mysql2_chef_gem/libraries/provider_mysql2_chef_gem_mysql.rb:16:in `block in <class:Mysql>'
==> default:       
==> default:       mysql_client_installation_package("default") do
==> default:         action [:create]
==> default:         retries 0
==> default:         retry_delay 2
==> default:         default_guard_interpreter :default
==> default:         declared_type :mysql_client
==> default:         cookbook_name "dbfit_test"
==> default:         version "5.6"
==> default:         major_version "5.6"
==> default:         package_name "mysql-community-client"
==> default:         package_version "5.6.29-2.el7"
==> default:       end
==> default:       
==> default:       Platform:
==> default:       ---------
==> default:       x86_64-linux
==> default:       
==> default: [2016-10-17T21:59:48+00:00] INFO: Running queued delayed notifications before re-raising exception
==> default:     
==> default:     ================================================================================
==> default:     Error executing action `install` on resource 'mysql2_chef_gem[default]'
==> default:     ================================================================================
==> default:     
==> default:     Chef::Exceptions::Package
==> default:     -------------------------
==> default:     mysql_client_installation_package[default] (/var/chef/cache/cookbooks/mysql2_chef_gem/libraries/provider_mysql2_chef_gem_mysql.rb line 16) had an error: Chef::Exceptions::Package: yum_package[mysql-community-client] (/var/chef/cache/cookbooks/mysql/libraries/mysql_client_installation_package.rb line 18) had an error: Chef::Exceptions::Package: Version 5.6.29-2.el7 of mysql-community-client not found. Did you specify both version and release? (version-release, e.g. 1.84-10.fc6)
==> default:     
==> default:     Resource Declaration:
==> default:     ---------------------
==> default:     # In /var/chef/cache/cookbooks/dbfit_test/recipes/mysql.rb
==> default:     
==> default:       3: mysql2_chef_gem 'default' do
==> default:       4:   gem_version '0.3.18'
==> default:       5:   client_version node['mysql']['version'] if node['mysql']
==> default:       6:   action :install
==> default:       7: end
==> default:       8: 
==> default:     
==> default:     Compiled Resource:
==> default:     ------------------
==> default:     # Declared in /var/chef/cache/cookbooks/dbfit_test/recipes/mysql.rb:3:in `from_file'
==> default:     
==> default:     mysql2_chef_gem("default") do
==> default:       action [:install]
==> default:       retries 0
==> default:       retry_delay 2
==> default:       default_guard_interpreter :default
==> default:       declared_type :mysql2_chef_gem
==> default:       cookbook_name "dbfit_test"
==> default:       recipe_name "mysql"
==> default:       gem_version "0.3.18"
==> default:     end
==> default:     
==> default:     Platform:
==> default:     ---------
==> default:     x86_64-linux
==> default:     
==> default: [2016-10-17T21:59:48+00:00] INFO: Running queued delayed notifications before re-raising exception
==> default: 
==> default: Running handlers:
==> default: [2016-10-17T21:59:48+00:00] ERROR: Running exception handlers
==> default: Running handlers complete
==> default: [2016-10-17T21:59:48+00:00] ERROR: Exception handlers complete
==> default: Chef Client failed. 1 resources updated in 14 seconds
==> default: [2016-10-17T21:59:48+00:00] FATAL: Stacktrace dumped to /var/chef/cache/chef-stacktrace.out
==> default: [2016-10-17T21:59:48+00:00] FATAL: Please provide the contents of the stacktrace.out file if you file a bug report
==> default: [2016-10-17T21:59:48+00:00] ERROR: mysql2_chef_gem[default] (dbfit_test::mysql line 3) had an error: Chef::Exceptions::Package: mysql_client_installation_package[default] (/var/chef/cache/cookbooks/mysql2_chef_gem/libraries/provider_mysql2_chef_gem_mysql.rb line 16) had an error: Chef::Exceptions::Package: yum_package[mysql-community-client] (/var/chef/cache/cookbooks/mysql/libraries/mysql_client_installation_package.rb line 18) had an error: Chef::Exceptions::Package: Version 5.6.29-2.el7 of mysql-community-client not found. Did you specify both version and release? (version-release, e.g. 1.84-10.fc6)
==> default: [2016-10-17T21:59:48+00:00] FATAL: Chef::Exceptions::ChildConvergeError: Chef run process exited unsuccessfully (exit code 1)
Chef never successfully completed! Any errors should be visible in the
output above. Please fix your recipes so that they properly complete.

**Comment by MMatten** [comment]
> Also in the same README: As of the 8.0 version of this cookbook, configuration of the package repos is now the responsibility of the user. We used to be on version 6.1.0 as far as I see from Cheffile.lock - so maybe some changes are needed when migrating to v8+.

Aha. Great.

So we need to be able to configure the yum repo on the guest first. Do you know how we can automate that? The RPM packages here (http://dev.mysql.com/downloads/repo/yum/) to configure yum seem to need to be selected according to the required version then manually downloaded and installed.

**Comment by javornikolov** [comment]
Regarding the yum repo: let's try with manual setup first, to make cookbooks pass. We can think of automation later when the other problems are sorted out.

About the package version - perhaps the chef cookbook defaults are for older release of CentOS/RHEL with a default of 5.6. Try to explicitly set the version in Vagrantfile. There is a line in `vendor/cookbooks/dbfit_test/recipes/mysql.rb`:
``` ruby
mysql2_chef_gem 'default' do
  gem_version '0.3.18'
  client_version node['mysql']['version'] if node['mysql']
  action :install
end
```

the idea is to influence it to pick `5.7` explicitly.

**Comment by MMatten** [comment]
> BTW, I can see recipe[selinux::disabled] mentioned as requirement here: https://github.com/chef-cookbooks/mysql

For this, can I just add `include_recipe "selinux::disabled"` in the `default.rb` recipe of the `dbfit_test` cookbook (after adding the `depends          'selinux'` in the `metadata.rb`)?

**Comment by javornikolov** [comment]
I think yes (about selinux recipe). But that's also something which can be manually disabled initially.

**Comment by javornikolov** [comment]
Default versions of mysql packages seem to be specified in that file: `https://github.com/chef-cookbooks/mysql/blob/master/libraries/helpers.rb`

**Comment by javornikolov** [comment]
Some stuff within its own tests:
[.kitchen.yml](https://github.com/chef-cookbooks/mysql/blob/fdd8e7d2317aab7e303797a33a6b2bfafa4b258e/.kitchen.yml)
[est/cookbooks/mysql_test/recipes/yum_repo.rb](https://github.com/chef-cookbooks/mysql/blob/fdd8e7d2317aab7e303797a33a6b2bfafa4b258e/test/cookbooks/mysql_test/recipes/yum_repo.rb)

Do you see `yum-mysql-community` in your Cheffile.lock file?

**Comment by MMatten** [comment]
> About the package version - perhaps the chef cookbook defaults are for older release of CentOS/RHEL with a default of 5.6. Try to explicitly set the version in Vagrantfile. There is a line in vendor/cookbooks/dbfit_test/recipes/mysql.rb:

I'm not exactly sure what I should poke. Can you give me something to try?

**Comment by MMatten** [comment]
> Do you see yum-mysql-community in your Cheffile.lock file?

Nope. The content is: -

```
SITE
  remote: http://community.opscode.com/api/v1
  specs:
    apt (5.0.0)
      compat_resource (>= 12.14.7)
    build-essential (7.0.1)
      compat_resource (>= 12.14)
      mingw (>= 1.1)
      seven_zip (>= 0.0.0)
    chef-sugar (3.4.0)
    compat_resource (12.14.7)
    database (6.0.0)
      postgresql (>= 1.0.0)
    dmg (3.0.0)
    firewalld (1.1.1)
    git (5.0.1)
      build-essential (>= 0.0.0)
      dmg (>= 0.0.0)
      yum-epel (>= 0.0.0)
    homebrew (2.1.2)
      build-essential (>= 2.1.2)
    java (1.42.0)
      apt (>= 0.0.0)
      homebrew (>= 0.0.0)
      windows (>= 0.0.0)
    lvm (3.0.0)
    mariadb (0.3.1)
      apt (>= 0.0.0)
      yum (>= 0.0.0)
      yum-epel (>= 0.0.0)
    mingw (1.2.4)
      compat_resource (>= 0.0.0)
      seven_zip (>= 0.0.0)
    mysql (8.0.4)
    mysql2_chef_gem (1.1.0)
      build-essential (>= 0.0.0)
      mariadb (>= 0.0.0)
      mysql (>= 6.0)
    openssl (4.0.0)
      chef-sugar (>= 0.0.0)
    selinux (0.9.0)
    seven_zip (2.0.2)
      windows (>= 1.2.2)
    windows (2.0.2)
    yum (4.0.0)
    yum-epel (1.0.1)
      yum (>= 3.6)

GIT
  remote: https://github.com/dbfit/postgresql
  ref: develop
  sha: 8c6a3847478f1723700858816c8b78fa8e4cf2d9
  specs:
    postgresql (3.4.21)
      apt (>= 1.9.0)
      build-essential (>= 0.0.0)
      openssl (~> 4.0.0)

GIT
  remote: https://github.com/dbfit/swap
  ref: master
  sha: 579982b444ac81256470915f20c04197081eae3f
  specs:
    swap (0.3.8)

PATH
  remote: vendor/cookbooks/dbfit_test
  specs:
    dbfit_test (0.2.0)
      database (>= 0.0.0)
      firewalld (>= 0.0.0)
      git (>= 0.0.0)
      java (>= 0.0.0)
      lvm (>= 0.0.0)
      mysql (>= 0.0.0)
      mysql2_chef_gem (>= 0.0.0)
      postgresql (>= 0.0.0)
      selinux (>= 0.0.0)
      swap (>= 0.0.0)
      windows (>= 1.2.2)
    windows (2.0.2)
    yum (4.0.0)
    yum-epel (1.0.1)
      yum (>= 3.6)

GIT
  remote: https://github.com/dbfit/postgresql
  ref: develop
  sha: 8c6a3847478f1723700858816c8b78fa8e4cf2d9
  specs:
    postgresql (3.4.21)
      apt (>= 1.9.0)
      build-essential (>= 0.0.0)
      openssl (~> 4.0.0)

GIT
  remote: https://github.com/dbfit/swap
  ref: master
  sha: 579982b444ac81256470915f20c04197081eae3f
  specs:
    swap (0.3.8)

PATH
  remote: vendor/cookbooks/dbfit_test
  specs:
    dbfit_test (0.2.0)
      database (>= 0.0.0)
      firewalld (>= 0.0.0)
      git (>= 0.0.0)
      java (>= 0.0.0)
      lvm (>= 0.0.0)
      mysql (>= 0.0.0)
      mysql2_chef_gem (>= 0.0.0)
      postgresql (>= 0.0.0)
      selinux (>= 0.0.0)
      swap (>= 0.0.0)
```

**Comment by javornikolov** [comment]
OK... I think that `manual setup of the yum repos` is exactly related to removing the built-in dependency on yum-mysql-community (hence you don't have it in Cheffile.lock anymore).
I think the related change has been in the [mysql cookbok](https://github.com/chef-cookbooks/mysql) - commit 824e854a1260c02800304a23c7d34efa8a627672 - `configure_package_repositories` has been removed, containing things like:
``` ruby
include_recipe('yum-mysql-community::mysql57'
```

> I'm not exactly sure what I should poke. Can you give me something to try?

I was thinking about Vagrantfile:
      "mysql" => {
        "server_root_password" => "password",
        "server_repl_password" => "password",
        "server_debian_password" => "password",
        "bind_address" => "127.0.0.1",
        **"version" => "5.7",**
      },

**Comment by MMatten** [comment]
Thanks. I thought that was it.

I'm now getting lib version conflicts. Probably with the MySQL client libs I tried installing manually earlier. I continue tomorrow with a refresh build.

**Comment by MMatten** [comment]
Ok, so now I'm hitting sinfomicien/mysql2_chef_gem#22.

I've got the MySQL yum repo configured on the guest using the `yum-mysql-community` cookbook. For now I've just included the `yum-mysql-community::mysql57` recipe in the main `default.rb` (`dbfit_test` cookbook).

I've tried adding `package 'mysql-community-devel'` into the start of our `mysql.rb`. I can't see the `mysql-devel` or `libmysqlclient-dev` packages on the MySQL yum repo.

Alas, now I'm hitting:

```
==> default: STDERR: 
==> default: 
==> default: Transaction check error:
==> default:   file /usr/bin/mysql_config-64 from install of mysql-community-devel-5.7.16-1.el7.x86_64 conflicts with file from package mysql-community-client-5.7.11-1.el7.x86_64
```

**Comment by javornikolov** [comment]
Maybe somehow you already installed one version of mysql (manually or some of the recepies), and then it tries to install a different version...

**Comment by MMatten** [comment]
I read that CentOS 7 comes with MariaDB pre-installed. I wonder if that's so and if that's conflicting.

But it didn't seem to be an issue on the previous CentOS 7 boxes we've used.

I'll see what I can find on the base box.

**Comment by MMatten** [comment]
I wonder if some packages are being pulled from the official MySQL repo and some from CentOS.

It's pretty confusing. I'm somewhat stuck.

**Comment by javornikolov** [comment]
> I've tried adding package 'mysql-community-devel' into the start of our mysql.rb.

Is that still there? Maybe we don't need that step. Are there any dependencies which won't be met if we remove that?

What do you think about publishing a pull request with your tweaks so far? (To have a better picture of the current state).

A few notes which I hope could be helpful to debugging such troubles in general:

* I think there is no mysql in CentOS & RHEL 7 own repos.
* The setup of yum repos is usually done within `/etc/yum.repos.d/`. The chef cookbooks are changing these - I don't recall if that change is permanent or just temporary for the execution.
* Seems `yum-mysql-community` is using the Oracle repo by default [`mysql57-community.rb`](https://github.com/chef-cookbooks/yum-mysql-community/blob/889360d3ed80ede6dfeefff74ba261be7bfb5ade/attributes/mysql57-community.rb)
* If you download .rpm and check it's internals (http://dev.mysql.com/downloads/repo/yum/) - it can be seen see the official yum url of the MySQL. E.g. for 5.7 it is: http://repo.mysql.com/yum/mysql-5.7-community/el/7/x86_64/
  * Navigating to the url can be seen that packages of multiple 5.7 versions are available.

There are some ways to capture logs/debug info from the chef provisioning. E.g. maybe these hints could be useful:
- https://www.vagrantup.com/docs/provisioning/chef_common.html
- https://mlafeldt.github.io/blog/debugging-chef-runs-with-chef-log/

**Comment by javornikolov** [comment]
If for some reason `mysql-community-devel` is needed and not automatically pulled by the other recipes - I think it's possible to specify more specific version (or version-release) in `yum_package`: https://docs.chef.io/resource_yum.html

**Comment by MMatten** [comment]
OK. It's hanging together now and I've drafted #513 to make it visible for refinement.

**Comment by MMatten** [comment]
Note that I've not yet tied adding Oracle / DB2 / Informix.

**Comment by MMatten** [comment]
@javornikolov any reason for us to keep this issue open still or shall I close it down?

**Comment by javornikolov** [comment]
You can close it, I believe it has been resolved with #513
